bitcoin/src/crypto/sighash.rs

```diff
@@ -1071,9 +1071,9 @@ impl<R: Borrow<Transaction>> SighashCache<R> {
         self.segwit_cache.get_or_insert_with(|| {
             let common_cache = Self::common_cache_minimal_borrow(common_cache, tx);
             SegwitCache {
-                prevouts: common_cache.prevouts.hash_again(),
-                sequences: common_cache.sequences.hash_again(),
-                outputs: common_cache.outputs.hash_again(),
+                prevouts: sha256d::Hash::from_hash(common_cache.prevouts),
+                sequences: sha256d::Hash::from_hash(common_cache.sequences),
+                outputs: sha256d::Hash::from_hash(common_cache.outputs),
```

Comment:
Why? I find the previous way of writing much more clear in conveying sha256 -> sha256d conversion. Perhaps `from_single` would work too.

Reply:
Agreed. `from_hash` sounds like it's converting from some more-fundamental hash type, which for sha256d, there is none.

Reply:
Done, thanks.